os/binfmt/libelf/libelf_load.c

```diff
@@ -157,11 +155,9 @@ static void elf_elfsize(struct elf_loadinfo_s *loadinfo)
 
 	/* Save the allocation size */
 #ifdef CONFIG_APP_BINARY_SEPARATION
-	loadinfo->textsize = 1 << mpu_log2regionceil(0, textsize);
-	loadinfo->rosize = 1 << mpu_log2regionceil(0, rosize);
-#else
-	loadinfo->textsize = textsize;
+	loadinfo->rosize = rosize;
```

Comment:
Maybe, this is also CONFIG_OPTIMIZE_APP_RELOAD_TIME.

Reply:
Yes.. I have changed it. 
I have also made some more related changes in other files to bring the ro section related calculations under the CONFIG_OPTIMIZE_APP_RELOAD_TIME flag.